packages/@react-aria/overlays/src/usePreventScroll.ts

```diff
@@ -46,10 +51,11 @@ export function usePreventScroll(options: PreventScrollOptions = {}) {
       return;
     }
```

Comment:
I think we could maybe simplify this by just having a global boolean that tracks whether prevent scroll is active, and if `usePreventScroll` is called again, just do nothing. It is always applied to the document body, so applying it again should be a no-op.

Reply:
Lemme give that a shot. Just thinking out loud, wouldn't using a global boolean be insufficient information since we want to prevent the document body prevent scroll style clean up when the menu is closed since the modal Dialog is open at that time?

Reply:
Just tried it, ran into the problem outlined above where the scroll style clean up happens when the menu is closed but the modal Dialog is open. I think we need have to track the number of active overlays (i.e. the number of preventScroll calls) so that we can tell the final overlay that is closed to perform the scroll style cleanup.

Reply:
Why would the cleanup happen if you return early and don't return the cleanup function from the useLayoutEffect?

Reply:
The cleanup function for the original overlay (aka menu) is the one that is cleaning up the scroll styles right? The flow that would happen is:
1. Menu mounts -> `usePreventScroll` is called, `isScrollPrevented` is set to true.
2. Dialog from menu item mounts -> `usePreventScroll` is called, we early return since `isScrollPrevented` is called so we any of the `preventScroll` functions
3. Menu unmounts -> cleanup from `preventScrollStandard`'s `setStyles` is called which re-enabled scrolling. However the Dialog is still open so we really shouldn't be re-enabling scrolling

Maybe I'm misunderstanding something here, but ideally we'd prevent the style cleanup when the menu unmounts since there is another overlay open.

Reply:
Oh right, yeah you'd need a counter. You could store the whole cleanup function in a global as well, and when the counter reaches zero, call it.

Reply:
> You could store the whole cleanup function in a global as well

I don't think I quite follow, is the ask here to extract the cleanup logic currently being done in `preventScrollMobileSafari` and `preventScrollStandard` out into a global cleanup function? Or is it to move
```
    if (isIOS()) {
      return chain(() => preventScrollCount--,  preventScrollMobileSafari());
    } else {
      return chain(() => preventScrollCount--,  preventScrollStandard());
    }
```
into a global cleanup just for some additional clarity in usePreventScroll?

Reply:
Something like this

```js
let preventScrollCount = 0;
let restore;

/// ...

useLayoutEffect(() => {
  preventScrollCount++;
  if (preventScrollCount === 1) {
    if (isIOS()) {
      restore = preventScrollMobileSafari();
    } else {
      restore = preventScrollStandard();
    }
  }

  return () => {
    preventScrollCount--;
    if (preventScrollCount === 0) {
      restore();
    }
  }
});
```

Reply:
Ahhh, I see. That makes sense, removes the need for storing the original body style values in a map.